perun-core/src/main/java/cz/metacentrum/perun/core/bl/MembersManagerBl.java

```diff
@@ -1854,4 +1857,31 @@ List<Map<String, String>> createSponsoredMembersFromCSV(PerunSession sess, Vo vo
 	 * @throws NamespaceRulesNotExistsException if there are no namespace rules for the given namespace
 	 */
 	void checkSponsoredUserData(PerunSession sess, SponsoredUserData data) throws InvalidSponsoredUserDataException, NamespaceRulesNotExistsException;
+
+	/**
+	 * Update value of memberOrganizations or memberOrganizationsHistory attribute for specific member.
+	 *
+	 * @param sess session
+	 * @param vo VO
+	 * @param member member
+	 * @param attributeName name of updated attribute
+	 * @throws WrongAttributeValueException if the attribute value is illegal
+	 * @throws WrongReferenceAttributeValueException if attribute which is reference for used attribute has illegal value
+	 * @throws AttributeNotExistsException  if the attribute doesn't exists in the underlying data source
+	 * @throws WrongAttributeAssignmentException if attribute is not vo attribute
+	 */
+	void updateOrganizationsAttribute(PerunSession sess, Vo vo, Member member, String attributeName) throws AttributeNotExistsException, WrongAttributeAssignmentException, WrongAttributeValueException, WrongReferenceAttributeValueException;
```

Comment:
Yes, but the current values of memberOrganization and memberOrganizationsHistory can be different (right?), so I can not set both attributes to the same value. Without attributeName, I will duplicate code, or I can make this method private and call it for both attributes from some new public function which will take only sess, vo, and member.

Reply:
Well, you could make a list of the attribute names and iterate through it to avoid code duplication. Either way, you probably want to use the constants A_MEMBER_DEF_MEMBER_ORGANIZATIONS_HISTORY and A_MEMBER_DEF_MEMBER_ORGANIZATIONS instead of hardcoding the attribute names. I still believe a single method for updating both attributes at the same time would be more convenient.